crates/llm-base/src/inference_session.rs

```diff
@@ -225,6 +228,67 @@ impl InferenceSession {
         Ok(stats)
     }
 
+    /// Calculate perplexity over a given prompt.
+    pub fn perplexity(
+        &mut self,
+        model: &dyn Model,
+        parameters: &InferenceParameters,
+        request: &InferenceRequest,
+        output_request: &mut OutputRequest,
+    ) {
+        // Setup the output request to store logits.
+        output_request.all_logits = Some(Vec::new());
+        let mut prompt_tokens = model
```

Comment:
Hm, is this the only use of `request`? It could be worth taking a `&[TokenId]` instead and letting users decide if they want to use `InferenceSession::tokens` or pass in their own tokens (potentially with a hint about tokenization)

Reply:
Like adding a cli for passing tokens to `--perplexity` that are different from `-p`?

Reply:
Ah, don't worry about it for now, this is fine :)